moduleroot/.github/workflows/ci.yml.erb

```
@@ -0,0 +1,106 @@
+name: CI
+
+on:
+  pull_request:
+  schedule:
+    - cron: '4 4 * * *'
+
+jobs:
+  setup_matrix:
+    if: github.event_name != 'schedule' || github.repository_owner == '<%= @configs[:namespace] %>'
+    name: 'Setup Test Matrix'
+    runs-on: ubuntu-latest
+    outputs:
+      beaker_setfiles: ${{ steps.get-outputs.outputs.beaker_setfiles }}
+      puppet_major_versions: ${{ steps.get-outputs.outputs.puppet_major_versions }}
+      puppet_unit_test_matrix: ${{ steps.get-outputs.outputs.puppet_unit_test_matrix }}
+    env:
+      BUNDLE_WITHOUT: development:release
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
+          bundler-cache: true
+      - name: Run rake check
+        run: bundle exec rake check
+      - name: Run rake validate
+        run: bundle exec rake validate
+      - name: Run rake lint
+        run: bundle exec rake lint
+      - name: Setup Test Matrix
+        id: get-outputs
+        run: bundle exec metadata2gha --use-fqdn --pidfile-workaround <%= @configs['pidfile_workaround'] %>
+
+  unit:
+    if: github.event_name != 'schedule' || github.repository_owner == '<%= @configs[:namespace] %>'
+    needs: setup_matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{fromJson(needs.setup_matrix.outputs.puppet_unit_test_matrix)}}
+    env:
+      BUNDLE_WITHOUT: development:system_tests:release
+      PUPPET_VERSION: "${{ matrix.puppet }}.0"
+    name: Puppet ${{ matrix.puppet }} (Ruby ${{ matrix.ruby }})
```

Given the whole CI file has `name: CI`, this now generates the following jobs:
- CI / Puppet 6 (Ruby 2.5) 
- CI / Puppet 5 (Ruby 2.4)

And the acceptance block below adds
- CI / Puppet 6 - CentOS 7
- CI / Puppet 5 - CentOS 7
- CI / Puppet 6 - CentOS 8
- CI / Puppet 5 - CentOS 8

Compared to the old output:
- Unit tests / Puppet 6 (Ruby 2.5)
- Unit tests / Puppet 5 (Ruby 2.4)
- Acceptance tests / Puppet 6 - centos7-64{hostname=centos7-64.example.com}
- Acceptance tests / Puppet 5 - centos7-64{hostname=centos7-64.example.com}
- Acceptance tests / Puppet 6 - centos8-64{hostname=centos8-64.example.com}
- Acceptance tests / Puppet 5 - centos8-64{hostname=centos8-64.example.com}

This makes it less obvious which kind of test failed, so I would propose

```suggestion
    name: Unit: Puppet ${{ matrix.puppet }} (Ruby ${{ matrix.ruby }})
```

or

```suggestion
    name: Puppet ${{ matrix.puppet }} (Ruby ${{ matrix.ruby }}) (Unit tests)
```

to make it more obvious from the overview what failed